packages/model-viewer/src/three-components/CachingGLTFLoader.ts

```diff
@@ -83,6 +87,19 @@ export class CachingGLTFLoader<T extends GLTFInstanceConstructor =
     return ktx2TranscoderLocation;
   }
 
+  static setMeshoptDecoderLocation(url: string) {
+    if (meshoptDecoderLocation !== url) {
+      meshoptDecoderLocation = url;
+      meshoptDecoder = fetchScript(url)
+        .then(() => MeshoptDecoder.ready)
+        .then(() => MeshoptDecoder);
```

Comment:
We usually use async/await instead of .then; would that be reasonable here too?

Reply:
I think we need to store a Promise, so additional invocations of this method don't make another request. Doing this with async/await would look something like:

```js
meshoptDecoder = new Promise(async (resolve, reject) => {
  try {
    await fetchScript(url);
    await MeshoptDecoder.ready;
    resolve(MeshoptDecoder);
  } catch (e) {
    reject(e);
  }
});
```

A bit more verbose, but could be wrapped in a helper function too, any preference here?

Reply:
Thanks for  the explanation; I'm fine with it as is.